integr8ly/integreatly-operator
# MGDAPI-4907 - Add MCG reconciler

# Issue link
[MGDAPI-4907](https://issues.redhat.com/browse/MGDAPI-4907)

# What
- Adds MCG reconciler
	- Installs MCG operator from index into mcg-operator namespace
	- Creates CR's for NooBaa system and ObjectBucketClaim
	- Populates 3scale s3-credentials from the created bucket
	- Uninstall removes CRs, the operator and the namespace
	- Unit testing above

Not yet completed - this could be done as separate ticket as we are merging to gcp branch:
- Disable MCG installation on non-GCP cluster

# Verification steps

Happy to pair with reviewer on verification, but here are the steps to do it yourself.

Provision a GCP CCS cluster - if you do not have an access key,  request one from myself - place in root of delorean repo and run:
```
OCM_CLUSTER_NAME=acatterm-ccs OCM_CLUSTER_LIFESPAN=24 COMPUTE_NODES_COUNT=4 BYOC=true CLOUD_PROVIDER=gcp make -f make/ocm.mk ocm/cluster.json
make -f make/ocm.mk ocm/cluster/create
```

Once provisioned we can deploy RHOAM from this branch:
```
LOCAL=false make cluster/prepare/local
LOCAL=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
LOCAL=false make code/run
```

You should see installation progress, and the new mcg product in RHMI CR.

Once install has completed, verify that the s3-credentials secret in redhat-rhoam-3scale namespace is as expected - the values from the following env vars should be present.

```
export AWS_ACCESS_KEY_ID=$(oc get secret 3scale-operator-bucket-claim -n 3scale-operator -o yaml | yq '.data.AWS_ACCESS_KEY_ID | @base64d')
export AWS_SECRET_ACCESS_KEY=$(oc get secret 3scale-operator-bucket-claim -n redhat-rhoam-mcg-operator -o yaml | yq '.data.AWS_SECRET_ACCESS_KEY | @base64d')
export AWS_BUCKET_NAME=$(oc get objectbucketclaim 3scale-operator-bucket-claim -n redhat-rhoam-mcg-operator -o yaml | yq '.spec.bucketName')
export AWS_ENDPOINT=$(oc get route s3 -n redhat-rhoam-mcg-operator -o yaml | yq '.spec.host')
```

You can then login to 3scale admin portal and check that the developer portal is accessible. To confirm everything is working as intended, you can also test the upload of an image in the developer portal settings. Once uploaded you can check that the item exists in the bucket.

```
aws --endpoint https://$AWS_ENDPOINT --no-verify-ssl s3api list-objects --bucket $AWS_BUCKET_NAME
```
